tests/integration/release/test_release.py

```diff
@@ -153,6 +154,62 @@ def test_full_release_noising(
                         fuzzy_checker,
                     )
 
+    # post-processing tests on final data
+    for dataset in datasets:
+        # these functions are called by Dataset during as part of noising process
+        # after noise types have been applied
+        dataset.data = coerce_dtypes(dataset.data, dataset.dataset_schema)
+        dataset.data = Dataset.drop_non_schema_columns(dataset.data, dataset.dataset_schema)
+
+        test_column_dtypes(dataset.data)
+    # do this outside loop to avoid reading data multiple times
+    test_unnoised_id_cols(datasets, dataset.dataset_schema.name)
+
+
+def test_column_dtypes(
+    noised_data: pd.DataFrame,
+) -> None:
+    """Tests that column dtypes are as expected"""
+    for col_name in noised_data.columns:
+        col = COLUMNS.get_column(col_name)
+        expected_dtype = col.dtype_name
+        if expected_dtype == np.dtype(object):
+            # str dtype is 'object'
+            # Check that they are actually strings and not some other
+            # type of object.
+            # mypy wants typed type_function to pass into apply but doesn't
+            # accept type as an output
+            type_function: Callable[..., Any] = lambda x: type(x)
+            actual_types = noised_data[col.name].dropna().apply(type_function)
+            with check:
+                assert (actual_types == str).all(), actual_types.unique()
+        with check:
+            assert noised_data[col.name].dtype == expected_dtype
+
+
+def test_unnoised_id_cols(datasets: list[Dataset], dataset_name: str) -> None:
```

Comment:
This was updated to take in list of Datasets instead of a DataFrame through a fixture.